core/src/Kokkos_Macros.hpp

```diff
@@ -228,8 +228,8 @@
 #define KOKKOS_ENABLE_PRAGMA_SIMD 1
 #endif
 
-// FIXME Workaround for ICE with intel 17,18,19 in Trilinos
-#if (KOKKOS_COMPILER_INTEL <= 1900)
+// FIXME Workaround for ICE with intel 17,18,19,20,21 in Trilinos
+#if (KOKKOS_COMPILER_INTEL <= 2100)
```

Comment:
Not objecting to the change but wondering if that actually means for all Intel C++ compilers currently released.

Reply:
`Intel(R) oneAPI DPC++/C++ Compiler 2022.1.0 (2022.1.0.20220316` has `__INTEL_LLVM_COMPILER=20220100`.

Reply:
I haven't tested with the 2022 compilers, it's possible the issue is present for that version as well